[GH_16717/MM_20389] Migrate 'components/signup/signup_controller' module and associated tests to TypeScript

aggmoulik opened the pull request:
Signed-off-by: Moulik Aggarwal <qwertymoulik@gmail.com>

<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute

REMEMBER TO:
- Run `make check-style` to check for style errors (required for all pull requests)
- Run `make test` to ensure unit tests passed
- Run `make check-types` to ensure type checking passed
- Add or update unit tests (required for all new features)
- All new/modified APIs include changes to the [JavaScript driver](https://github.com/mattermost/mattermost-redux/blob/master/src/client/client4.js)
-->

#### Summary
It changes the name of the SubscriptionStats in camelcase as it was written in bad mannered and added a property which is used in the react web app.

#### Ticket Link
Fixes: https://mattermost.atlassian.net/browse/MM-20389

mattermod commented:
Hello @aggmoulik,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

devinbinnie commented on src/types/cloud.ts:
```diff
@@ -110,6 +110,6 @@ export type InvoiceLineItem = {
 export type SubscriptionStats = {
-	remaining_seats: number;
-	is_paid_tier: string;
+	remainingSeats: number;
```
We can't change the names of these without changing what's being passed down from the server, since it's mapped using snake case naming. I think these names will need to be changed on the webapp PR instead.

aggmoulik replied:
I created a PR for the same on the web-app repository and if you want me to remove these naming convention changes then it will not pass the type-checking.

lieut-data replied:
Agree with @devinbinnie -- until we go to the effort of renaming all the "incoming" fields from the server, we're stuck with the snack case here. Can we supply instructions to override the linter in this case?

devinbinnie replied:
Just to confirm @aggmoulik, you're saying the linter is causing the error or the type checker? If it's the type checker, there's probably another instance that needs to be changed somewhere as well.

If it's the linter, then I think we need to change our ruleset to allow for snake case unless we plan on changing this convention app-wide.

hmhealey replied:
Having the server code return snake case was intentional way back when, so I don't think there's any plan to change that any time soon.

I'd say we could update the ESLint rules for that now, but we're going to be getting rid of mattermost-redux as a separate repository later this week, so I'd say to leave the casing as-is and use an `// eslint-ignore` wherever the warnings are thrown for the time being.

devinbinnie reviewed (changes requested):
Added a comment for you to review @aggmoulik.
Thanks for the contribution!

aggmoulik commented:
@lieut-data I can do it in the same PR of webapp.

aggmoulik commented:
Hy @lieut-data @devinbinnie , I have removed the naming in redux and also fixed the web-app PR. So, please review this so that web-app PR can pass all checks.

devinbinnie reviewed (approved):
LGTM!

lieut-data reviewed (changes requested):
Thanks, @aggmoulik! Just one comment below.

lieut-data commented:
Weird, my comment didn't actually show up -- the ask was to sort the new entry alphabetically with the rest.

aggmoulik commented:
> Weird, my comment didn't actually show up -- the ask was to sort the new entry alphabetically with the rest.

Updated @lieut-data

lieut-data reviewed (approved):
Thanks!!
